apiary.apib

```diff
@@ -586,4 +586,232 @@ Creates a filled form
+              "href": "/users{?id}",
+              "templated": true
+            }
+          },
+          "_embedded": {
+            "ea:user": [
+              {
+                "_links": {
+                  "self": {
+                    "href": "/users/123"
+                  }
+                },
+                "id": 123,
+                "email":"jane.shepard@normandysr2.gov",
+                "active": "true",
+                "confirmed_at": "2182-04-11T18:25:43Z",
+                "last_login_at": "2183-09-19T13:11:33Z",
+                "current_login_at": "2183-09-19T13:11:33Z",
+                "last_login_ip": "50.41.123.231",
+                "current_login_ip": "50.41.123.231",
+                "login_count": 22,
+                "name": "Jane Shephard",
+                "title": "Commander,
+              },
+            ]
+          }
+        }
+
+
+## GET /users/{id}
+Gets a user by id
++ Parameters
+    + id (required, nuber) ... The unique id
++ Response 200
+
+{
+  "_links": {
+    "self": {
+      "href": "/users/123"
+    }
+  },
+  "id": 123,
+  "email":"jane.shepard@normandysr2.gov",
+  "active": "true",
+  "confirmed_at": "2182-04-11T18:25:43Z",
+  "last_login_at": "2183-09-19T13:11:33Z",
+  "current_login_at": "2183-09-19T13:11:33Z",
+  "last_login_ip": "50.41.123.231",
+  "current_login_ip": "50.41.123.231",
+  "login_count": 22,
+  "name": "Jane Shephard",
+  "title": "Commander,
+}
+
+
+# Users CRUD [/users]
+
+## POST /users
+Creates a new user
++ Parameters
+    + email (required, string) ... The user's email address
+    + password (required, string) ... the user's password
+    + active (optional, string) ... Whether the account should be active
+
+        + Values
+            + `true`
+            + `false`
+
+    + secret (optional, string) ...
+    + name (optional, string) ... User's real-life name
+    + title (optional, string) ... User's job title
++ Request
+
+        {
+        "email":"jane.shepard@normandysr2.gov",
+        "password":"15omnigel",
+        }
++ Response 201
+
+        {
+          "_links": {
+            "self": "/users/123",
+            }
+        "id": 123,
+        "email": "jane.shepard@normandysr2.gov",
+        "active": "true",
+        "confirmed_at": null,
+        "last_login_at": null,
+        "current_login_at": null,
+        "last_login_ip": null,
+        "current_login_ip": null,
+        "login_count": 0,
+        "name": null,
+        "title": null,
+        }
+
+
+## PUT /users/{id}
+Replaces existing user record.  Any fields not supplied are filled with `null`.
++ Parameters
+    + id (required, integer) ... Existing record identifier
+    + email (required, string) ... The user's email address
+    + password (required, string) ... the user's password
+    + active (optional, string) ... Whether the account should be active
+
+        + Values
+            + `true`
+            + `false`
+
+    + secret (optional, string) ...
```

Comment:
I believe -- but could be wrong -- that the `secret` field in the database contains the user's current authentication token.